### ISSUE TITLE
Retrieval Error returns: The received offer errored: retrieval query offer errored: datastore: key not found

### ISSUE THREAD

**Issue by stefangeorgerobert** [filecoin-project/lotus#2314]
**Describe the bug**
lotus client retrieve fails on lotus version 0.4.1+git.5de7b62d.dirty, I've reproduced the issue with multiple files / miners. 

**To Reproduce**
**lotus client import small.txt** 
bafkreigzngt27rdxdw24qq73nt7w5kohtcku6vtmrddgwurjjtty25te2i
**lotus client deal bafkreigzngt27rdxdw24qq73nt7w5kohtcku6vtmrddgwurjjtty25te2i t0118155 0.0000000005 10000**
bafyreiegbzwosui5p5byv4kcgu7apa2nevsat2u6es57iigvac2fitkewa
**lotus client list-deals | grep bafyreiegbzwosui5p5byv4kcgu7apa2nevsat2u6es57iigvac2fitkewa**
bafyreiegbzwosui5p5byv4kcgu7apa2nevsat2u6es57iigvac2fitkewa  4512  t0118155  StorageDealActive        Y (epoch 64667)  N  bafk4chzamd2ychkzcnovi4dppvlmteacutbezt4qez5vr6imbgi4kmrupuma  254 B      500000000             16288 

**lotus client find bafkreigzngt27rdxdw24qq73nt7w5kohtcku6vtmrddgwurjjtty25te2i**
LOCAL
ERR t0116061@12D3KooWMhn1bm4oQpiTn8Z8KN7cAWU4rYfE3AT3Rf7KcumeAL9f: retrieval query offer errored: get cid info: No state for /bafkreigzngt27rdxdw24qq73nt7w5kohtcku6vtmrddgwurjjtty25te2i: datastore: key not found

**lotus client retrieve bafkreigzngt27rdxdw24qq73nt7w5kohtcku6vtmrddgwurjjtty25te2i outfile1.txt**
2020-07-08T08:14:35.650Z	WARN	main	lotus/main.go:81	The received offer errored: retrieval query offer errored: get cid info: No state for /bafkreigzngt27rdxdw24qq73nt7w5kohtcku6vtmrddgwurjjtty25te2i: datastore: key not found

**Expected behavior**
lotus client retrieve command should retrieve the file.

**Screenshots**

**Version (run `lotus version`):**
lotus version 0.4.1+git.5de7b62d.dirty

**Additional context**

**Comment by flyworker** [comment]
I have the same issue as well

**lotus client list-deals**
DealCid                                                      DealId  Provider  State                    On Chain?        Slashed?  PieceCID                                                       Size       Price      Duration  Message
bafyreiearfkac5r7hmwi3wxcapu5oje3orwaoysiqgd4qgm47d6qepgnrm  4648    t021341   StorageDealActive        Y (epoch 68877)  N         bafk4chzat5vtqnxbeiv747alu7ag3syrj5tfmnqfczwr2sh44n4hh3rdnaea  254 MiB    600000000  33579

**lotus client retrieve bafkreibsjrokruayihzp624ybk7kttoauvorazhuimj6t53zj77pez6qjm 0.jpg**
2020-07-09T00:44:59.133Z	WARN	main	lotus/main.go:81	The received offer errored: retrieval query offer errored: get cid info: No state for /bafkreibsjrokruayihzp624ybk7kttoauvorazhuimj6t53zj77pez6qjm: datastore: key not found

**Comment by hannahhoward** [comment]
@flyworker retrieval is done by the data CID for what was stored, NOT the deal CID

You appear to be using the DealCID.

**Comment by panol** [comment]
Same problem on interop.6.16.0

**Comment by flyworker** [comment]
> @flyworker retrieval is done by the data CID for what was stored, NOT the deal CID
> 
> You appear to be using the DealCID.
> 
> The correct CID to use for retrieval is the one you get when you do `lotus client import` -- see how @stefangeorgerobert did it above.
> 
> I am not sure what the issue @stefangeorgerobert is having means.
I find the out of the one I got from import 

```
$lotus client import  ./m0vp4DxU_400x400.jpg
bafkreibsjrokruayihzp624ybk7kttoauvorazhuimj6t53zj77pez6qjm
```
and

```
$ lotus client retrieve bafkreibsjrokruayihzp624ybk7kttoauvorazhuimj6t53zj77pez6qjm  0.jpg
2020-07-10T05:49:42.521Z	WARN	main	lotus/main.go:81	The received offer errored: retrieval query offer errored: get cid info: No state for /bafkreibsjrokruayihzp624ybk7kttoauvorazhuimj6t53zj77pez6qjm: datastore: key not found
```

**Comment by mishmosh** [comment]
@stefangeorgerobert Thanks for the writeup. Just to confirm, are you seeing this when running branch `next`?

**Comment by arajasek** [comment]
*May* be fixed by #2372

**Comment by TippyFlitsUK** [comment]
Addressed in https://github.com/filecoin-project/lotus/pull/2372
